### DISCUSSION TITLE
issue: Mui checkbox will not dirty on first attempt in production build

### DISCUSSION THREAD

**Discussion by domshyra** [react-hook-form/react-hook-form#10280] [Q&A] [answered]
### Version Number

7.43.9

### Codesandbox/Expo snack

https://github.com/domshyra/ReactHookFormsBug

This project has a few branches on it too that were tested and still resulted in the same place as the `main` branch. The readme of those branches should outline the approach in that specific branch. 

### Steps to reproduce

I created a sample application that has the error in it. 

When running in production build, the form will only dirty on a second field being touched. The expected behavior is how it runs in local dev with the form being dirty on the first change. 

### Expected behavior

When a use clicks show modal for the first time and they hit a checkbox, the save button should be enabled from the first checkbox click. This works with the radio buttons and text field but not the checkbox one. 

### What browsers are you seeing the problem on?

Chrome, Firefox, Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
Please follow the following todos:

- build a simple codesandbox focus on the issue
- deploy a prod version of that csb

**Comment by domshyra** [comment]
[Here is the codesandbox](https://codesandbox.io/p/sandbox/elastic-booth-xf4ylh?welcome=true&file=%2FREADME.md)

3000 is the local build
4000 is the prod build 

The prod version might need to be refreshed after the first time using it. This seems to only happen on the very first checkbox interaction when built with prod. 

Edit: link fixed

**Comment by bluebill1049** [comment]
<img width="900" alt="Screenshot 2023-04-15 at 8 24 37 am" src="https://user-images.githubusercontent.com/10513364/232164754-d82d4e80-ce88-4cce-a610-56d1f4de62e3.png">
check your link first.

**Comment by domshyra** [comment]
Ah apologies, link fixed

**Comment by bluebill1049** [comment]
Please remove all external dependencies and isolated to only react hook form and the MUI component with which you having your issue.

**Comment by leapful** [comment, marked as answer]
@domshyra 

Please follow carefully the documentation of `formState`

<img width="1023" alt="Screenshot 2023-04-16 at 13 53 06" src="https://user-images.githubusercontent.com/18494222/232278411-dbc68af3-7cfd-4bd4-b816-bb69af4b4e0d.png">

You must read `formState` as it's a Proxy before rendering to enable status update. In development with StrictMode enable, component might be rendered twice so this issue won't happen. 

Reference: https://react-hook-form.com/api/useform/formstate/

Working demo: https://codesandbox.io/p/sandbox/keen-monad-pvmrsl?file=%2Fweb%2Fsrc%2Fcomponents%2FModals%2FModalForForms.jsx
